03-rnaseq/gene-id-annotation_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,382 @@
+  # We only need a preview here
+  head()
+```
+
+You may have a list of Entrez IDs you are interested in, in which case, the above format may work best for you.
+In a different study, you may want the oldest Entrez ID (which is probably first), in which case, you can create a column that stores just the first mapped Entrez ID that comes up for each Ensembl ID.
+We will do this in the next section by re-running the `mapIds()` function with `multiVals = "first"`.
+
+### Map Ensembl IDs to gene symbols -- keeping only the first mapped value
+
+If we don't have a particular preference of which Entrez ID is returned, we can have `mapIds()` use its default of returning the first Entrez ID listed.
+
+Let's re-run `mapIds()`, this time using `multiVals = "first"`.
+
+```{r}
+final_mapped_df <- data.frame(
+  "first_mapped_entrez_id" = mapIds(
+    org.Dr.eg.db, # Replace with annotation package for the organism relevant to your data
+    keys = df$Gene,
+    column = "ENTREZID", # Replace with the type of gene identifiers you would like to map to
+    keytype = "ENSEMBL", # Replace with the type of gene identifiers in your data
+    multiVals = "first" # This will keep only the first mapped value for each Ensembl ID
+  )
+) %>%
+  # Make an `Ensembl` column to store the rownames
+  tibble::rownames_to_column("Ensembl") %>%
+  # Join the multiple mappings data from `collapsed_mapped_df` using the Ensembl IDs
+  dplyr::inner_join(collapsed_mapped_df, by = "Ensembl") %>%
+  # Now let's join the rest of the expression data
+  dplyr::inner_join(df, by = c("Ensembl" = "Gene"))
+```
+
+Our `final_mapped_df` object now has a column named `first_mapped_entrez_id` that contains only the first mapped Entrez ID, in addition to the `all_entrez_ids` column that contains all mapped Entrez IDs per Ensembl ID.
+
+```{r}
+final_mapped_df %>%
+  # Filter `final_mapped_df` to preview the rows where `all_entrez_ids` values include the ";" character -- we know that these are the rows with multiple mapped values
+  dplyr::filter(stringr::str_detect(all_entrez_ids, ";")) %>%
+  head()
+```
+
+Now let's write our mapped results and data to file!
+
+## Write mapped results to file
+
+```{r}
+# Write mapped and annotated data frame to output file
+readr::write_tsv(final_mapped_df, file.path(
+  results_dir,
+  "SRP040561_Entrez_IDs.tsv" # Replace with a relevant output file name
+))
+```
+
+# Resources for further learning
+
+- Marc Carlson has prepared a nice [Introduction to Bioconductor Annotation Packages](https://bioconductor.org/packages/release/bioc/vignettes/AnnotationDbi/inst/doc/IntroToAnnotationPackages.pdf) [@Carlson2020-vignette]
+- See our [microarray gene ID conversion notebook](https://github.com/AlexsLemonade/refinebio-examples/tree/master/02-microarray/gene-id-annotation_microarray_01_ensembl.html) as another applicable example, since the steps for this workflow do not change with technology [@gene-id-annotation-microarray].
```

Comment:
```suggestion
- See our [microarray gene ID conversion notebook](https://alexslemonade.github.io/refinebio-examples/02-microarray/gene-id-annotation_microarray_01_ensembl.html) as another applicable example, since the steps for this workflow do not change with technology [@gene-id-annotation-microarray].
```